src/components/nav_drawer/nav_drawer.tsx

```diff
@@ -215,11 +283,11 @@ export class EuiNavDrawer extends Component {
     }
   };
 
-  modifyChildren = children => {
+  modifyChildren = (children: ReactElement[]): ReactElement[] => {
```

Comment:
`children` could be any valid React node, not only elements. This applies to the return value as well.

```suggestion
  modifyChildren = (children: ReactNode): ReactNode => {
```